app/models/scc_account.rb

```diff
@@ -230,4 +254,32 @@ def update_scc_products(upstream_products)
       end
     end
   end
+
+  # validate the subscription status of a product/repo
+  # no saving to database
+  # params: elements: scc repos or products, Array or ActiveRecord_(*)
+  #         scc_id: scc_id of the element that should be revalidated
+  # return: elements where for the element with scc_id subscription_valid is true
+  def revalidate_subscription_status(elements, scc_id)
+    return nil if elements.nil?
+
+    revalidate = elements.find { |e| e.scc_id == scc_id }
+    revalidate.subscription_valid = true unless revalidate.nil?
+    # return modified list
+    elements
+  end
+
+  # set all products/repos invalid
+  # params: items_to_invalidate: ActiveRecord_(*)
+  #         save_record: store in database or not (default)
+  # return: ActiveRecord elements with invalidated subscription status
+  def invalidate_subscription_status(items_to_invalidate, save_record = false)
+    unless items_to_invalidate.empty?
+      items_to_invalidate.each do |inv|
+        inv.subscription_valid = false
+        inv.save! if save_record
+      end
```

Comment:
Might be faster to do it like this, because it will result in a single SQL UPDATE command.
```suggestion
      if save_record
        items_to_invalidate.update_all subscription_valid: false
      else
        items_to_invalidate.each { |inv| inv.subscription_valid = false }
      end
```

Reply:
I tried doing this and rubocop does not like it:
`scc_manager/app/models/scc_account.rb:279:29: C: Rails/SkipsModelValidations: Avoid using update_all because it skips validations.
        items_to_invalidate.update_all subscription_valid: false
`

Reply:
if we are _sure_ that it does not break validations, we can use it. otherwise, stay at the previoius method or use
inv.update(subscription_valid: false)

Reply:
The `update_all` skips validations. Shall I change it back the way it was?

Reply:
I would argue that flipping this boolean field does not affect any validations.

Though I am ok with using the original approach, given that this method is not called that often.

Reply:
The field is also allowed to be nil (e.g. after migration or when manually adding an SccProduct) and should be correct after the first SccAccount sync.

Reply:
I'm fine with leaving it as it originally was.